cluster/cluster.go

```diff
@@ -563,3 +580,117 @@ func MustGetSegmentConfiguration(connection *dbconn.DBConn, getMirrors ...bool)
+gpsegconfig_dump file gets created at $COORDINATOR_DATA_DIR/gpsegconfig_dump by fts process
+The gpsegconfig_dump file follows a structured format, as illustrated in the example below:
+1 -1 p p n u 6000 localhost localhost /data/temp1
+2 0 p p n u 6002 localhost localhost /data/temp2
+3 1 p p n u 6003 localhost localhost /data/temp3
+4 2 p p n u 6004 localhost localhost /data/temp4
+
+Recommended use of the api is to populate the gpdbArray when database is down.
+if the database is up use GetSegmentConfiguration()/MustGetSegmentConfiguration()  API.
+
+note: since the gpsegconfig_dump file is updated by fts process the information returned by
+this function can be a bit stale since user can configure fts to run less frequently
+
+input:
+coordinator_data_dir - coordinator data dir path, can be retrieved from env var COORDINATOR_DATA_DIRECTORY
+(e.g. /Users/shrakesh/workspace/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1)
+
+return:
+Slice of populated SegConfig objects or else nil
+error if any or else nil.*/
+
+func GetSegmentConfigurationFromFile(coordinatorDataDir string) ([]SegConfig, error) {
+
+	/*Check if the given argument coordinator_data_dir is empty*/
+	if len(strings.TrimSpace(coordinatorDataDir)) == 0 {
+		errStr := fmt.Sprintf("Error: Given coordinator data directory name is empty")
+		return nil, fmt.Errorf(errStr)
+	}
+
+	/*Generate gpsegconfig_dump file path*/
+	gpsegconfigDump := path.Join(coordinatorDataDir, "gpsegconfig_dump")
+
+	/* Open gpsegconfig_dump */
+	fd, err := os.Open(gpsegconfigDump)
+	if err != nil {
+		errStr := fmt.Sprintf("Failed to open file %s. Error: %s", gpsegconfigDump, err.Error())
+		return nil, fmt.Errorf(errStr)
+	}
+	defer fd.Close()
+
+	results := make([]SegConfig, 0)
+	scanner := bufio.NewScanner(fd)
+
+	/* Anonymous function to convert provided field from string to int */
+	convertField := func(fieldName string, fieldValue string) (int, error) {
+		val, err := strconv.Atoi(fieldValue)
+		if err != nil {
+			errStr := fmt.Sprintf("Failed to convert %s with value %s to an int. Error: %s", fieldName, fieldValue, err.Error())
+			return -math.MaxInt, fmt.Errorf(errStr)
+		}
+		return val, nil
+	}
+
+	/*scanning file line by line to extract the fields into SegConfig struct*/
+	for scanner.Scan() {
+		fields := strings.Fields(scanner.Text())
+		parts := len(fields)
+		var datadir string
+
+		/* older version of gpsegconfig_dump has 9 parts as it doesn't have datadir
+			1 -1 p p n u 7000 shrakeshSMD6M.vmware.com shrakeshSMD6M.vmware.com
+		newer version of gpsegconfig_dump has 10 parts as it does have datadir
+			1 -1 p p n u 7000 shrakeshSMD6M.vmware.com shrakeshSMD6M.vmware.com /data/qddir/demoDataDir-1 */
+		if parts != 9 && parts != 10 {
+			errStr := fmt.Sprintf("Unexpected number of fields (%d) in line: %s", parts, scanner.Text())
+			return nil, fmt.Errorf(errStr)
+		}
+
+		dbID, err := convertField("dbID", fields[0])
```

Comment:
Now that we aren't logging the error using gplog, we might not need this function since it's a simple oneliner now

```
dbID, err := strconv.Atoi("dbID")
if err != nil {
	return nil, fmt.Sprintf("Failed to convert dbid with value %s to an int. Error: %s", ...)
}
```